gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/adapter/PortalNavigationItemAdapter.java

```diff
@@ -0,0 +1,161 @@
+        expression = "java(portalNavigationItem.getParentId() != null ? PortalNavigationItemId.of(portalNavigationItem.getParentId()) : null)"
+    )
+    @Mapping(target = "order", source = "order")
+    @Mapping(target = "href", expression = "java(parseHref(portalNavigationItem.getConfiguration()))")
+    PortalNavigationLink portalNavigationLinkFromRepository(
+        io.gravitee.repository.management.model.PortalNavigationItem portalNavigationItem
+    );
+
+    @Mapping(target = "id", expression = "java(PortalNavigationItemId.of(portalNavigationItem.getId()))")
+    @Mapping(target = "organizationId", source = "organizationId")
+    @Mapping(target = "environmentId", source = "environmentId")
+    @Mapping(target = "title", source = "title")
+    @Mapping(target = "area", expression = "java(mapArea(portalNavigationItem.getArea()))")
+    @Mapping(
+        target = "parentId",
+        expression = "java(portalNavigationItem.getParentId() != null ? PortalNavigationItemId.of(portalNavigationItem.getParentId()) : null)"
+    )
+    @Mapping(target = "order", source = "order")
+    @Mapping(target = "contentId", expression = "java(PortalPageContentId.of(parsePageId(portalNavigationItem.getConfiguration())))")
+    PortalNavigationPage portalNavigationPageFromRepository(
+        io.gravitee.repository.management.model.PortalNavigationItem portalNavigationItem
+    );
+
+    @Mapping(target = "id", expression = "java(portalNavigationItem.getId().json())")
+    @Mapping(
+        target = "parentId",
+        expression = "java(portalNavigationItem.getParentId() != null ? portalNavigationItem.getParentId().json() : null)"
+    )
+    @Mapping(target = "type", expression = "java(mapType(portalNavigationItem))")
+    @Mapping(target = "area", expression = "java(mapAreaReverse(portalNavigationItem.getArea()))")
+    @Mapping(target = "configuration", expression = "java(configurationOf(portalNavigationItem))")
+    io.gravitee.repository.management.model.PortalNavigationItem toRepository(PortalNavigationItem portalNavigationItem);
+
+    default io.gravitee.repository.management.model.PortalNavigationItem.Type mapType(PortalNavigationItem portalNavigationItem) {
+        return switch (portalNavigationItem) {
+            case PortalNavigationFolder ignored -> io.gravitee.repository.management.model.PortalNavigationItem.Type.FOLDER;
+            case PortalNavigationPage ignored -> io.gravitee.repository.management.model.PortalNavigationItem.Type.PAGE;
+            case PortalNavigationLink ignored -> io.gravitee.repository.management.model.PortalNavigationItem.Type.LINK;
+        };
+    }
+
+    default String configurationOf(PortalNavigationItem portalNavigationItem) {
+        try {
+            return switch (portalNavigationItem) {
+                case PortalNavigationPage page -> {
+                    Map<String, String> config = new HashMap<>();
+                    config.put("pageId", page.getContentId().json());
```

Comment:
To avoid confusion in the future, can you use `portalPageContentId` everywhere ?